packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -78,4 +97,97 @@ export class ApplicationExtension<
         const {allRoutes} = params
         return allRoutes
     }
+
+    /**
+     * Called on the server to serialize the extension data that will be sent to the client.
+     *
+     * @returns SerializedExtension - The serialized extension data.
+     * @throws Error if the routes have not been loaded.
+     */
+    public serialize(): SerializedExtension {
+        if (this._cachedRoutes === null) {
+            throw new Error(`Routes have not been loaded. Call getRoutesAsync() before serializing`)
+        }
+        console.log('--- serializing routes for extension', this.getName(), this._cachedRoutes)
+        const serializedRoutes = this._cachedRoutes.map((route) => {
+            // Check if the route is already serialized
+            if ('componentName' in route) {
```

Comment:
That checks whether `route` has the key `componentName`. It doesn't care about its _value_.

So if `route.componentName = undefined`, `'componentName' in route` will still return true.

Let's do this instead to verify the value:
```suggestion
            if (route.componentName) {
```